### ISSUE TITLE
Warn user if target drive is mounted

### ISSUE THREAD

**Issue by CyanEbi** [calamares/calamares#2306]
**Is your feature request related to a problem? Please describe.**
In the EndeavourOS Galileo Neo 2024.01.25 live-environment I had mounted the drive that I intended to install EOS on (in order to backup the files on it), and I didn't think to unmount the drive afterwards. When reaching the _Partitions_ section of Calamares I notice that only _Manual partitioning_ is available:
![Calamares_only_manual](https://github.com/calamares/calamares/assets/159061101/8e87c439-209b-4c2d-8813-e96c0af5eb48)
After defining the partitions and reaching the _Install_ section, the installation fails with the following error displayed:
![Calamares_failed](https://github.com/calamares/calamares/assets/159061101/c4badd5f-7478-47dc-9313-1a9d8e696d2e)
Starting again with the target drive unmounted the expected partitioning options appear and the installer runs as expected:
![Calamares_all_options](https://github.com/calamares/calamares/assets/159061101/b374cbaa-975d-4b11-8932-197c157611f0)

**Describe the solution you'd like**
I think this situation can be avoided if Calamares checks whether the selected drive has mounted partitions in the _Partitions_ section - and if so warns the user that they may want to unmount those to get all partitioning options (possibly even adding a button to unmount the selected drive)

**Describe alternatives you've considered**

**Additional context**

**Comment by demmm** [comment]
Please post the full installation log, so it can be seen why the normal unmounting is not applied.

**Comment by CyanEbi** [comment]
The installation succeeded after I had unmounted the drive so I couldn't recreate the state the drive was in, but I made an attempt by formatting the drive back to a single ext4 partition like it was when the problem occurred. Unfortunately I couldn't reproduce the error - the _Partitions_ section still only showed _Manual partitioning_, but the installation ran without issue this time. 
The issue may have been a consequence of my previous OS (on a neighbouring drive) being a mess and possibly messing with the other drives.
Sorry for letting that bug get away.

However I would still suggest warning users about mounted partitions to avoid confusion about why only _Manual partitioning_ is available
